book/security.md

```diff
@@ -847,23 +1045,34 @@ The server is much simpler, but has also grown since last chapter:
 Exercises
 =========
 
-*Hidden input*: Add support for hidden input elements. Since they
-don't need to be laid out, there's no need for a layout type at all!
+*New inputs*: Add support for hidden and password input elements.
+Hidden inputs shouldn't show up or take up space, while password input
+elements should show ther contents as stars instead of characters.
+
+*Cookie access*: Add support for the `HttpOnly` and `Secure`
+parameters for cookies. `HttpOnly` cookies should not be available
+from JavaScript's `document.cookie`, but should be sent along with
+HTTP requests. `Secure` cookies should only be sent over HTTPS, but
+not over HTTP, requests.  `HttpOnly` cookies provide some protection
+against cross-site scripting attacks, while `Secure` cookies prevent
+an attacker from reading cookies sent over unencrypted networks.
 
 *Cookie Expiration*: Add support for cookie expiration. Cookie
 expiration dates are set in the `Set-Cookie` header, and can be
-overwritten if the same cookie is set again with a later date. Save
-the same expiration dates in the `TOKENS` variable and use it to
-delete old tokens to save memory.
-
-*Cookie Origins*: Add support for cookie origins. Due to the
-same-origin policy, a cookie set by `mail.google.com` cannot be read
-by, say, `calendar.google.com`, because the host name is the same.
-This is a good default,[^23] but is sometimes annoying. A server can
-set an `origin` parameter in the `Set-Cookie` header to strip off some
-subdomains from the cookie origin. Implement this in your browser,
-making sure to send these generalized-origin cookies on any requests
-covered by the generalized origin.
+overwritten if the same cookie is set again with a later date. On the
+server side, save the same expiration dates in the `SESSIONS` variable
+and use it to delete old sessions to save memory.
+
+*CORS*: Web servers can [*opt in*](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) to allowing cross-origin
+`XMLHttpRequest`s using the `Access-Control-Allow-Origin` header. The
+way it works is that on cross-origin HTTP requests, the web browser
+sends an `Origin` header with the origin of the requesting site. By
+default, the browser then throws away the response to prevent private
+data from leaking. But if the server sends the
+`Access-Control-Allow-Origin` header, and its value is either the
+requesting origin or the special `*` value, the browser instead makes
+the output available to the script. All requests made by your browser
+will be what the CORS standard calls "simple requests".
 
 *Referer*: When your browser visits a web page, or when it loads a CSS
 or JavaScript file, it sends a `Referer` header[^24] containing the
```

Comment:
```suggestion
or JavaScript file, it sends a `Referer` header[^referer] containing the
```

Reply:
Thanks. Yep, the initial footnotes were auto-exported from Org-mode, and it's good to use descriptive names as we edit.